### ISSUE TITLE
Any way to stop response generation after one sentence?

### ISSUE THREAD

**Issue by MrReplikant** [cloveranon/Clover-Edition#139]
Maybe not the best way to go about asking, but I was wondering if anyone here knows what would have to be changed in the generator py  in order to make the game generate single-sentence replies. Basically I just want it to stop after the first "?", ".", Or "!".  I've tried a few different things with stop tokens, but it was a fruitless effort. It should be noted I'm a bit of a beginner. Any help would be greatly appreciated!

**Comment by MikkoMMM** [comment]
You definitely can at least hack it in. I looked into this since I'm also planning on using this kind of functionality at some point.

The file you are going to want to modify is gpt2generator.py, and I'd suggest between lines 162 and 163 in it (in the outer sample_sequence function).

```
            last_token = tokenizer.decode(o[-1], clean_up_tokenization_spaces=False, skip_special_tokens=True)
            if ([ele for ele in [".", "!", "?"] if (ele in last_token)]):
                break
```

I think that last_token thing might be faster than searching through the entire sentence generated thus far (generated.text), even though the speed of such an operation hardly matters at all.

Looking at just generated.text[-1] was another interesting idea I had. Usually those characters are standalone tokens, and this is why that would work. However, there are edge cases where this is not the case, so I had to make something more robust. Feel free to close this issue if you're satisfied.
